src/configure/helper/commonHelper.ts

```diff
@@ -63,4 +67,36 @@ export function stringCompareFunction(a: string, b: string): number {
         return 1;
     }
     return 0;
+}
+
+export async function executeFunctionWithRetry(func: () => Promise<any>, retryCount: number = 100, retryTimeoutInSec: number = 300, retryIntervalTimeInSec: number = 2, errorMessage?: string): Promise<any> {
+    let internalErrorMessage = '';
+    let functionExecutionPromise = Q.Promise(async (resolve, reject) => {
+        let result = null;
+        for (;retryCount > 0; retryCount--) {
+            try {
+                result = await func();
+                resolve(result);
+                break;
+            }
+            catch (error) {
+                internalErrorMessage = errorMessage ? errorMessage.concat(' Internal Error: ', JSON.stringify(error)) : JSON.stringify(error);
+                setTimeout(() => {
+                    // do nothing
+                },
+                retryIntervalTimeInSec * 1000);
+            }
+        }
+
+        if (retryCount <= 0 && !result) {
+            reject(internalErrorMessage);
+        }
+    });
+
+    return await Q.timeout(functionExecutionPromise, retryTimeoutInSec * 1000, 'Timed out')
```

Comment:
Why do we need a timeout? Isn't retry count sufficient?

Reply:
Incase of slow reponse (rest call timeouts -> 100 seconds), the total time of retries can be very huge

Reply:
We will hardly run into this situation. I would prefer to keep it simple for now.

Reply:
okay will remove the timeout part